src/modules/otlphttp.cpp

```diff
@@ -0,0 +1,905 @@
+    return;
+  }
+  int cmetric_len = strlen(cmetric);
+  /* We will append the time stamp afer the null terminator to keep the key
+   * appropriately unique.
+   */
+  uint64_t t = w.tv_sec * 1000 + w.tv_usec / 1000;
+  memcpy(cmetric + cmetric_len + 1, &t, sizeof(uint64_t));
+  cmetric_len += 1 + sizeof(uint64_t);
+
+  if(mtev_hash_size(rxc->immediate_metrics) > 1000) {
+    metric_local_batch_flush_immediate(rxc);
+    return;
+  }
+  if(mtev_hash_retrieve(rxc->immediate_metrics, cmetric, cmetric_len, &vm)) {
+    /* collision, just log it out */
+    metric_local_batch_flush_immediate(rxc);
+    return;
+  }
+  //metric_t *m = mtev_memory_safe_malloc_cleanup(sizeof(*m), metric_local_free);
+  metric_t *m = static_cast<metric_t *>(malloc(sizeof(*m)));
+  memset(m, 0, sizeof(*m));
+  m->metric_name = mtev_strndup(cmetric, cmetric_len);
+  if(val) {
+    m->metric_type = METRIC_DOUBLE;
+    memcpy(&m->whence, &w, sizeof(struct timeval));
+    m->metric_value.vp = malloc(sizeof(double));
+    *(m->metric_value.n) = *val;
+  } else if(vali) {
+    m->metric_type = METRIC_INT64;
+    memcpy(&m->whence, &w, sizeof(struct timeval));
+    m->metric_value.vp = malloc(sizeof(int64_t));
+    *(m->metric_value.n) = *vali;
+  } else {
+    assert(val || vali);
+  }
+
+  noit_stats_mark_metric_logged(noit_check_get_stats_inprogress(rxc->check), m, mtev_false);
+  mtev_hash_store(rxc->immediate_metrics, m->metric_name, cmetric_len, m);
+}
+
+static int
+rest_otlphttp_handler(mtev_http_rest_closure_t *restc, int npats, char **pats)
+{
+  int mask, complete = 0, cnt;
+  otlphttp_upload_t *rxc = NULL;
+  const char *error = "internal error", *secret = NULL;
+  mtev_http_session_ctx *ctx = restc->http_ctx;
+  const unsigned int DEBUGDATA_OUT_SIZE=4096;
+  char debugdata_out[DEBUGDATA_OUT_SIZE];
+  int debugflag=0;
+  const char *debugchkflag;
+  noit_check_t *check;
+  uuid_t check_id;
+  mtev_http_request *req;
+  mtev_hash_table *hdrs;
+
+  if(npats != 2) {
+    error = "bad uri";
+    goto error;
+  }
+  if(mtev_uuid_parse(pats[0], check_id)) {
+    error = "uuid parse error";
+    goto error;
+  }
+
+  if(restc->call_closure == NULL) {
+    mtev_boolean allowed = mtev_false;
+    check = noit_poller_lookup(check_id);
+    if(!check) {
+      error = "no such check";
+      goto error;
+    }
+    if(strcmp(check->module, "otlphttp")) {
+      error = "no such otlphttp check";
+      goto error;
+    }
+
+    /* check "secret"  */
+    (void)mtev_hash_retr_str(check->config, "secret", strlen("secret"), &secret);
+    if(secret && !strcmp(pats[1], secret)) allowed = mtev_true;
+    if(!allowed && cross_module_reverse_allowed(check, pats[1])) allowed = mtev_true;
+
+    if(!allowed) {
+      error = "secret mismatch";
+      goto error;
+    }
+
+    rxc = new otlphttp_upload_t(check);
+    if(const char *mode_str = mtev_hash_dict_get(check->config, "hist_approx_mode"); mode_str) {
```

Comment:
I'm glad you're using this but in this case don't need the 3rd form of ``if``, the second form will do

```suggestion
    if(const char *mode_str = mtev_hash_dict_get(check->config, "hist_approx_mode")) {
```